src/Sentry/Laravel/Features/StorageIntegration.php

```diff
@@ -0,0 +1,83 @@
+<?php
+
+namespace Sentry\Laravel\Features;
+
+use Illuminate\Contracts\Filesystem\Cloud as CloudFilesystem;
+use Illuminate\Contracts\Filesystem\Filesystem;
+use Illuminate\Contracts\Foundation\Application;
+use Illuminate\Filesystem\FilesystemManager;
+use RuntimeException;
+use Sentry\Laravel\Tracing\Storage\TracingCloudFilesystem;
+use Sentry\Laravel\Tracing\Storage\TracingFilesystem;
+
+class StorageIntegration extends Feature
+{
+    private const FEATURE_KEY = 'storage';
+
+    private const STORAGE_DRIVER_NAME = 'sentry';
+
+    public function isApplicable(): bool
+    {
+        return $this->isBreadcrumbFeatureEnabled(self::FEATURE_KEY)
+               || $this->isTracingFeatureEnabled(self::FEATURE_KEY);
+    }
+
+    public function setup(): void
+    {
+        foreach (config('filesystems.disks') as $disk => $config) {
+            $currentDriver = $config['driver'];
+
+            if ($currentDriver === self::STORAGE_DRIVER_NAME) {
+                continue;
+            }
+
+            config([
+                "filesystems.disks.{$disk}.driver" => self::STORAGE_DRIVER_NAME,
+                "filesystems.disks.{$disk}.sentry_disk_name" => $disk,
+                "filesystems.disks.{$disk}.sentry_original_driver" => $config['driver'],
+            ]);
+        }
+
+        $recordSpans = $this->isTracingFeatureEnabled(self::FEATURE_KEY);
+        $recordBreadcrumbs = $this->isBreadcrumbFeatureEnabled(self::FEATURE_KEY);
```

Comment:
I think fetching this information can be done at a later point in the callback of `extend`. This avoids performing unnecessary work eagerly. Of course this relies on the callbacks only executing once, which I believe to be the case. I tried this by adding two `Storage` calls to a route and putting an `echo` in either place, both run only once.